### PR TITLE
Adds support for HomeAssistant Select Entity and Tuya Mode option 

### PR DESCRIPTION
**Pull request by Jhonattan-Souza**
Support for the new Home Assistant Entity introduced in the last update -> 

References:
https://www.home-assistant.io/blog/2021/07/07/release-20217/

https://developers.home-assistant.io/docs/core/entity/select

### CODE REVIEW

**Comment by Jhonattan-Souza** [comment]
@tsutsuku Can you review it for me? Thanks 🙏

File: custom_components/tuya_v2/select.py [hunk 1]
```diff
@@ -0,0 +1,88 @@
...
+
+    async_dispatcher_connect(
+        hass, TUYA_DISCOVERY_NEW.format(DEVICE_DOMAIN), async_discover_device
+    )
+
+    device_manager = hass.data[DOMAIN][TUYA_DEVICE_MANAGER]
+    device_ids = []
+    for (device_id, device) in device_manager.device_map.items():
+        if device.category in TUYA_SUPPORT_TYPE:
+            device_ids.append(device_id)
```
Comment (fabio-filho) [inline custom_components/tuya_v2/select.py:54]: The same here, what you think about to export this to a function called `get_supported_tuya_devices` ?

File: custom_components/tuya_v2/select.py [hunk 2]
```diff
@@ -0,0 +1,88 @@
...
+def _setup_entities(hass, device_ids: List):
+    device_manager = hass.data[DOMAIN][TUYA_DEVICE_MANAGER]
+    entities = []
+    for device_id in device_ids:
+        device = device_manager.device_map[device_id]
+        if device is None:
+            continue
+
+        if DPCODE_MODE in device.status:
+            entities.append(TuyaHaSelect(device, device_manager, DPCODE_MODE))
```
Comment (fabio-filho) [inline custom_components/tuya_v2/select.py:67]: What's the meaning of this "mode" value?

**Review by fabio-filho — COMMENTED** [review]
I just made some comments 😅

File: custom_components/tuya_v2/select.py (hunk 2 above)
Comment (tsutsuku) [reply custom_components/tuya_v2/select.py:67]: It's a DP [Data Point] Code.

**Review by tsutsuku — COMMENTED** [review]
Thank you for your contribution.

**Pull request merged**